breacharbiter.go

```diff
@@ -443,21 +523,30 @@ secondLevelCheck:
 	// channel's retribution against the cheating counter party.
 	err = b.cfg.PublishTransaction(finalTx)
 	if err != nil {
-		brarLog.Errorf("unable to broadcast "+
-			"justice tx: %v", err)
+		brarLog.Errorf("unable to broadcast justice tx: %v", err)
+
 		if err == lnwallet.ErrDoubleSpend {
-			brarLog.Infof("Attempting to transfer HTLC revocations " +
-				"to the second level")
+			// Broadcasting the transaction failed because of a
+			// conflict either in the mempool or in chain. We'll
+			// now create spend subscriptions for all HTLC outputs
+			// on the commitment transaction that could possibly
+			// have been spent, and wait for any of them to
+			// trigger.
+			brarLog.Infof("Waiting for a spend event before " +
+				"attempting to craft new justice tx.")
 			finalTx = nil
 
-			// Txn publication may fail if we're shutting down.
-			// If so, return to avoid entering an infinite loop.
-			select {
-			case <-b.quit:
+			err := b.waitForSpendEvent(breachInfo, spendNtfns)
+			if err != nil {
+				if err != errBrarShuttingDown {
+					brarLog.Errorf("error waiting for "+
+						"spend event: %v", err)
+				}
 				return
-			default:
-				goto secondLevelCheck
 			}
+
+			brarLog.Infof("Attempting another justice tx broadcast")
+			goto justiceTxBroadcast
 		}
 	}
```

Comment:
Should `return` here, otherwise we'll register for a confirmation on a transaction not published.

Reply:
It's a good point, but this keeps the existing behaviour, and it doesn't hurt to do it. This will let us wait for confirmation in the case where it _does_ get published, but an error is returned regardless.